chain/network/src/peer_manager/testonly.rs

```diff
@@ -0,0 +1,57 @@
+use crate::network_protocol::testonly as data;
+use crate::peer_manager::peer_manager_actor::Event as PME;
+use crate::testonly::actix::ActixSystem;
+use crate::testonly::fake_client;
+use crate::PeerManagerActor;
+use crate::Sink;
+use actix::Actor;
+use near_network_primitives::types::NetworkConfig;
+use near_store::test_utils::create_test_store;
+use std::sync::Arc;
+use tokio::sync::mpsc;
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum Event {
+    Client(fake_client::Event),
+    PeerManager(crate::peer_manager::peer_manager_actor::Event),
+}
+
+pub struct Handler {
+    pub cfg: NetworkConfig,
+    _actix: ActixSystem<PeerManagerActor>,
+    event_stream: mpsc::UnboundedReceiver<Event>,
+}
+
+impl Handler {
+    pub async fn recv(&mut self) -> Event {
+        self.event_stream.recv().await.unwrap()
+    }
+    // consume events, until Event::PeerManager happens.
+    pub async fn recv_peer_manager(&mut self) -> crate::peer_manager::peer_manager_actor::Event {
+        loop {
+            if let Event::PeerManager(ev) = self.recv().await {
+                return ev;
+            }
+        }
```

Comment:
Feels a bit odd that there's no graceful shutodown handling here. I'd expect standard

```
while let Ok(event) = reciever.recv()
```

loop which auto-closese when the sender is dropped.

Reply:
dropping the sender means that test fails (because we are waiting for some event). There is a panic within self.recv.

Reply:
Ah, right, this isn't an infinite loop, there's return there